### ISSUE TITLE
Pure CTC didn't early stop

### ISSUE THREAD

**Issue by wenjie-p** [espnet/espnet#3080]
Hi,

I recently found my model didn't early stop, after going through the code, I found there may be a typo here:

https://github.com/espnet/espnet/blob/f92aa89217ee69d5d89f46eadee145f304488a7b/espnet/utils/training/train_utils.py#L35

**Comment by b-flo** [comment]
Hi,

There is no typo, see class documentation: https://docs.chainer.org/en/stable/reference/generated/chainer.training.triggers.EarlyStoppingTrigger.html

> patients is also available as an alias of patience for historical reason.

**Comment by wenjie-p** [comment]
Oh sorry, thanks for the clarification!

> I recently found my model didn't early stop,

Actually I was running the pure CTC mode with VGGBLSTM model, but found it didn't early stop as the validation loss keeps increasing.

![image](https://user-images.githubusercontent.com/21860575/111063105-d7eedf00-84e7-11eb-882b-d22b093fbdbe.png)

I found the code:

https://github.com/espnet/espnet/blob/master/espnet/asr/pytorch_backend/asr.py#L800-L834

When using pure CTC mode, I think the `str` passed to `CompareValueTrigger` should be `validation/main/loss_ctc` rather than `validation/main/loss`. (in [line)](https://github.com/espnet/espnet/blob/master/espnet/asr/pytorch_backend/asr.py#L824)

The similar setting should also be changed for saving the best model:

https://github.com/espnet/espnet/blob/master/espnet/asr/pytorch_backend/asr.py#L781


I am sorry for didn't have extra machines to validate such assumption, but put it here first for discussion.

**Comment by wenjie-p** [comment]
The recipe I ran is `egs/commonvoice/asr1/` with config for rnn training:

```
etype: vggblstm     # encoder architecture type
elayers: 3
eunits: 1024
eprojs: 1024
subsample: "1_2_2_1_1" # skip every n frame from input to nth layers
# decoder related
dlayers: 2
dunits: 1024
# attention related
atype: location
adim: 1024
aconv-chans: 10
aconv-filts: 100

# hybrid CTC/attention
mtlalpha: 1.0
ctc-weight: 1.0

# minibatch related
batch-size: 32
maxlen-in: 800  # if input length  > maxlen_in, batchsize is automatically reduced
maxlen-out: 150 # if output length > maxlen_out, batchsize is automatically reduced

# optimization related
sortagrad: 0 # Feed samples from shortest to longest ; -1: enabled for all epochs, 0: disabled, other: enabled for 'other' epochs
opt: adadelta
epochs: 15
patience: 3

# scheduled sampling option
sampling-probability: 0.0
```

And patience = 3 in my exp.
